modules/aws-kernel/src-native/smithy4s/aws/CryptoPlatformCompat.scala

```diff
@@ -16,90 +16,64 @@
 
 package smithy4s.aws.kernel
 
-import scala.scalanative.unsafe
 import scala.scalanative.unsafe._
 import scala.scalanative.unsigned._
 
+import crypto._
+
 trait CryptoPlatformCompat {
   type Binary = Array[Byte]
 
+  private[this] val EvpMdSha256 = EVP_sha256()
+
   def binaryFromString(str: String): Binary =
     str.getBytes("UTF-8")
 
   def toHexString(binary: Binary): String =
     binary.map("%02x".format(_)).mkString
 
-  def sha256HexDigest(message: Array[Byte]): String =
-    unsafe.Zone { implicit z =>
-      val outLength: Ptr[CUnsignedInt] = unsafe.alloc[CUnsignedInt]()
-      val digest: Ptr[CUnsignedChar] =
-        unsafe.alloc[CUnsignedChar](CryptoPlatformCompat.EVP_MAX_MD_SIZE)
-      val dataLen = message.length.toLong
-      val cInput: CString = arrayToCString(message)
-
-      val ctx = crypto.EVP_MD_CTX_new()
-      crypto.EVP_DigestInit(ctx, crypto.EVP_sha256())
-      crypto.EVP_DigestUpdate(ctx, cInput, dataLen.toUInt)
-      crypto.EVP_DigestFinal(ctx, digest, outLength)
-
-      val out = Array.fill[Byte]((!outLength).toInt)(0)
-      var i = 0
-      while (i < (!outLength).toInt) {
-        out(i) = digest(i.toLong).toByte
-        i += 1
-      }
-
-      out.map("%02x".format(_)).mkString
-    }
+  def sha256HexDigest(message: Array[Byte]): String = {
+    val md = new Array[Byte](EVP_MAX_MD_SIZE)
+    val size = stackalloc[CUnsignedInt]()
+    if (
+      EVP_Digest(
```

Comment:
well I feel very dumb 😄 

Although, this reminds me of something I wanted to ask your opinion about : in the large majority of usecases, the AWS signature algorithm implies that the whole body of the outgoing http request lives in memory. 
Because that's the case, and because at this stage correctness matters much more than performance, I kinda went "well, might as well use byte arrays, duh". Without thinking twice about it, I did this : 

https://github.com/disneystreaming/smithy4s/blob/series/0.18/modules/aws-http4s/src/smithy4s/aws/internals/AwsSignatureClient.scala#L126

I'm wondering whether it'd be beneficial for the digest to be computed with knowledge of how the Chunk is structured, instead of just forcing a `.toArray` call. At the moment, the chunk is likely to be backed by a single `Array[Byte]`, but it may evolve in the future.

Reply:
> I'm wondering whether it'd be beneficial for the digest to be computed with knowledge of how the Chunk is structured

Hmm, since that's the http4s integration, couldn't we just use `fs2.hash.sha256` there?

Reply:
lol, good point